openmdao/core/problem.py

```diff
@@ -147,13 +220,33 @@ def __setitem__(self, name, value):
         value : float or ndarray or list
             value to set this variable to.
         """
-        if name in self.model._outputs:
-            self.model._outputs[name] = value
-        elif name in self.model._inputs:
-            self.model._inputs[name] = value
+        # Caching only needed if vectors aren't allocated yet.
+        if self._setup_status == 1:
+            self._initial_condition_cache[name] = value
         else:
-            msg = 'Variable name "{}" not found.'
-            raise KeyError(msg.format(name))
+            if name in self.model._outputs:
+                self.model._outputs[name] = value
+            elif name in self.model._inputs:
+                self.model._inputs[name] = value
+            else:
+                msg = 'Variable name "{}" not found.'
+                raise KeyError(msg.format(name))
+
+    def _set_initial_conditions(self):
+        """
+        Set all initial conditions that have been saved in cache after setup.
+        """
+        for name, value in iteritems(self._initial_condition_cache):
+            if name in self.model._outputs:
```

Comment:
I would have to set the setup_status to 2 a little earlier to do so, but I could do it.